WordPress/openverse
# Project thread reminders

## Fixes



Fixes #2250 by @zackkrida  

## Description



Adds a run-daily GitHub action which checks all project threads for comments in the last 14 days. If a thread hasn't been commented on, a comment is left on the issue asking the author to write an update. The comment is formatted like this:

Rendered:

Hi {recipient}, this project has not received an update comment in 14 days. Please leave an update comment as soon as you can. See the [documentation on project updates](https://docs.openverse.org/projects/planning.html#providing-project-updates) for more information.

In markdown:

```md
Hi @${recipient}, this project has not received an update comment in 14 days. Please leave an update comment as soon as you can. See the [documentation on project updates](https://docs.openverse.org/projects/planning.html#providing-project-updates) for more information.
```

## Testing Instructions


~This can be easily tested via a [manual workflow run](https://github.com/WordPress/openverse/actions/workflows/project_thread_update_reminders.yml). Be sure to leave the `Dry run` box checked otherwise the script will leave comments on several project threads 😄~

This testing mechanism has been removed. See the results below though!

You can see _my_ latest run and it's results [here](https://github.com/WordPress/openverse/actions/runs/5145365109/jobs/9262905496).

The threads identified to update by the script, as of today (2023-05-31), are:

- [ ] #394
- [ ] #392
- [ ] #412
- [ ] #410

Which should be correct. 

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
